packages/backend/src/telemetry/index.ts

```diff
@@ -0,0 +1,82 @@
+import { Config } from '../config/app'
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-grpc'
+import { Resource } from '@opentelemetry/resources'
+import {
+  MeterProvider,
+  PeriodicExportingMetricReader
+} from '@opentelemetry/sdk-metrics'
+
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc'
+import { api } from '@opentelemetry/sdk-node'
+import { PgInstrumentation } from '@opentelemetry/instrumentation-pg'
+import { GraphQLInstrumentation } from '@opentelemetry/instrumentation-graphql'
+
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
+import {
+  BatchSpanProcessor,
+  NodeTracerProvider
+} from '@opentelemetry/sdk-trace-node'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { UndiciInstrumentation } from '@opentelemetry/instrumentation-undici'
+
+// debug logger:
+// diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG)
+
+const SERVICE_NAME = 'RAFIKI_NETWORK'
+const rafikiResource = new Resource({
+  'service.name': SERVICE_NAME,
+  instance: Config.instanceName
+})
+
+const meterReaders = []
+
+if (Config.enableTelemetry) {
+  for (const url of Config.openTelemetryCollectors) {
+    const metricExporter = new PeriodicExportingMetricReader({
+      exporter: new OTLPMetricExporter({
+        url
+      }),
+      exportIntervalMillis: Config.openTelemetryExportInterval ?? 15000
+    })
+
+    meterReaders.push(metricExporter)
+  }
+}
+
+const meterProvider = new MeterProvider({
+  resource: rafikiResource,
+  readers: meterReaders
+})
+
+api.metrics.setGlobalMeterProvider(meterProvider)
+
+const tracerProvider = new NodeTracerProvider({
+  resource: rafikiResource
+})
+
+if (Config.enableTelemetryTraces) {
+  for (const url of Config.openTelemetryTraceCollectorUrls) {
+    const traceExporter = new OTLPTraceExporter({
+      url
+    })
+
+    tracerProvider.addSpanProcessor(new BatchSpanProcessor(traceExporter))
+  }
+}
+
+tracerProvider.register()
+
+registerInstrumentations({
+  tracerProvider,
+  meterProvider,
+  instrumentations: [
+    new UndiciInstrumentation(),
+    new HttpInstrumentation(),
+    new PgInstrumentation(),
+    new GraphQLInstrumentation({
+      mergeItems: true,
+      ignoreTrivialResolveSpans: true,
+      ignoreResolveSpans: true
+    })
```

Comment:
UndiciInstrumentation: Auto-instruments  node `fetch` module. Useful for us to track requests made with the open payments client, since it uses fetch under the hood.

HttpInstrumentation: Auto-instruments http and https modules. Useful to track client requests made with axios and server requests.

PgInstrumentation: auto-instruments all of our postgres queries.

GraphQLInstrumentation: auto-instruments our GraphQL APIs, particularly resolvers. The settings are to reduce the amount of noise we get. Otherwise, we see spans for every single field that we resolve. We can adjust if necessary.

Reply:
We are always instrumenting these, regardless of `enableTelemetryTraces`. Is that OK? I mean the important thing is that we're not sending them anywhere, so maybe it's fine. But I wonder if this could have a performance impact or cause any unnecessary concern.

Reply:
Although there shouldn't be much of a difference at all, it would probably be safer to put this behind the flag. 

I moved the code behind the flags, tested with them enabled & disabled, and everything seems to work. It also works if we make telemetry non optional and try to increment a counter while everything is disabled - it just does a noop.